samples/graph-notification-team-channel/csharp/GraphNotification/Helper/SubscriptionManager.cs

```diff
@@ -0,0 +1,364 @@
+        private readonly ILogger _logger;
+        private readonly GraphBetaClient graphBetaClientProvider;
+        public static readonly Dictionary<string, Subscription> Subscriptions = new Dictionary<string, Subscription>();
+        private string pageId;
+
+        public SubscriptionManager(IOptions<BotConfiguration> botSettings, ILogger<SubscriptionManager> logger, GraphBetaClient graphBetaClientProvider)
+        {
+            this.botSettings = botSettings;
+            this.graphBetaClientProvider = graphBetaClientProvider;
+            _logger = logger;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            await InitializeAllSubscription("", "");
+
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                await Task.Delay(TimeSpan.FromMinutes(SubscriptionRenewTimeInMinutes), stoppingToken).ConfigureAwait(false);
+                _logger.LogWarning("Renewal started.");
+                await this.CheckSubscriptions().ConfigureAwait(false); ;
+            }
+        }
+
+        public override async Task StartAsync(CancellationToken stoppingToken)
+        {
+            await InitializeAllSubscription("", "");
+
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                await Task.Delay(TimeSpan.FromMinutes(SubscriptionRenewTimeInMinutes), stoppingToken).ConfigureAwait(false);
+                _logger.LogWarning("Renewal started.");
+                await this.CheckSubscriptions().ConfigureAwait(false); ;
+            }
+        }
+
+        public async Task InitializeAllSubscription(string teamId, string pageId)
+        {
+            this.pageId = pageId;
+            _logger.LogWarning("InitializeAllSubscription-started");
+
+            if (pageId == "1")
+            {
+                await CreateNewSubscription(teamId);
+                await this.CheckSubscriptions().ConfigureAwait(false);
+            }
+            else
+            {
+                await CreateNewSubscriptionForTeam(teamId);
+                await this.CheckSubscriptionsForTeams().ConfigureAwait(false);
+            }
+            
+            _logger.LogWarning("InitializeAllSubscription-completed");
+        }
+
+        //channel
+        public async Task CheckSubscriptions()
+        {
+            _logger.LogWarning($"Checking subscriptions {DateTime.UtcNow.ToString("h:mm:ss.fff")}");
+
+            foreach (var subscription in Subscriptions)
+            {
+                await RenewSubscription(subscription.Value);
+            }
+        }
+        //teams
+        public async Task CheckSubscriptionsForTeams()
+        {
+            _logger.LogWarning($"Checking subscriptions {DateTime.UtcNow.ToString("h:mm:ss.fff")}");
+
+            foreach (var subscription in Subscriptions)
+            {
+                await RenewSubscriptionForTeams(subscription.Value);
+            }
+        }
+
```

Comment:
Add proper line spacing.
Also, add proper summary for methods.

Reply:
fixed